code/game/gamemodes/autotraitor/autotraitor.dm

```diff
@@ -56,128 +49,91 @@
-	spawn(9000)
-		if(SSshuttle.emergency.mode >= SHUTTLE_ESCAPE)
-			return
-		//message_admins("Performing AutoTraitor Check")
-		var/playercount = 0
-		var/traitorcount = 0
-		var/possible_traitors[0]
-		for(var/mob/living/player in GLOB.mob_list)
-			if(player.client && player.stat != DEAD)
-				playercount += 1
-				if(!player.mind)
-					continue
-				if(player.mind.special_role)
-					traitorcount += 1
-					continue
-				if(ishuman(player) || isAI(player))
-					if((ROLE_TRAITOR in player.client.prefs.be_special) && !player.client.skip_antag && !jobban_isbanned(player, ROLE_TRAITOR) && !jobban_isbanned(player, ROLE_SYNDICATE))
-						possible_traitors += player.mind
-		for(var/datum/mind/player in possible_traitors)
-			for(var/job in restricted_jobs)
-				if(player.assigned_role == job)
-					possible_traitors -= player
-			if(!player.current || !ishuman(player.current)) // Remove mindshield-implanted mobs from the list
-				continue
-			var/mob/living/carbon/human/H = player.current
-			for(var/obj/item/bio_chip/mindshield/I in H.contents)
-				if(I && I.implanted)
-					possible_traitors -= player
-			if(!H.job || H.mind.offstation_role) //Golems, special events stuff, etc.
-				possible_traitors -= player
-		//message_admins("Live Players: [playercount]")
-		//message_admins("Live Traitors: [traitorcount]")
-//		message_admins("Potential Traitors:")
-//		for(var/mob/living/traitorlist in possible_traitors)
-//			message_admins("[traitorlist.real_name]")
-
-//		var/r = rand(5)
-//		var/target_traitors = 1
-		var/max_traitors = 1
-		var/traitor_prob = 0
-		max_traitors = round(playercount / 10) + 1
-		traitor_prob = (playercount - (max_traitors - 1) * 10) * 5
-		if(traitorcount < max_traitors - 1)
-			traitor_prob += 50
-
-
-		if(traitorcount < max_traitors)
-			//message_admins("Number of Traitors is below maximum.  Rolling for new Traitor.")
-			//message_admins("The probability of a new traitor is [traitor_prob]%")
+	addtimer(CALLBACK(src, PROC_REF(traitor_check_loop)), 15 MINUTES)
 
-			if(prob(traitor_prob))
-				message_admins("Making a new Traitor.")
-				if(!length(possible_traitors))
-					message_admins("No potential traitors.  Cancelling new traitor.")
-					traitorcheckloop()
-					return
-				var/datum/mind/newtraitormind = pick(possible_traitors)
-				var/mob/living/newtraitor = newtraitormind.current
-				//message_admins("[newtraitor.real_name] is the new Traitor.")
-
-				to_chat(newtraitor, "<span class='danger'>ATTENTION:</span> It is time to pay your debt to the Syndicate...")
-				newtraitor.mind.add_antag_datum(/datum/antagonist/traitor)
-			//else
-				//message_admins("No new traitor being added.")
-		//else
-			//message_admins("Number of Traitors is at maximum.  Not making a new Traitor.")
-
-		traitorcheckloop()
+/datum/game_mode/traitor/autotraitor/proc/traitor_check_loop()
+	if(SSshuttle.emergency.mode >= SHUTTLE_ESCAPE)
+		return
 
+	var/player_count = 0
+	var/traitor_count = 0
+	var/possible_traitors[0]
```

Comment:
```suggestion
	var/list/possible_traitors = list()
```
I think this is just a quite fucked way to make a list?

Reply:
I missed that. Yeah, it doesn't look great.